packages/core/lib/BusIndexed.ts

```diff
@@ -69,8 +75,13 @@ export class BusIndexed<A extends Actor<I, T, O>, I extends IAction, T extends I
     return super.publish(action);
   }
 
-  protected getActorIdentifier(actor: A): string {
-    return this.actorIdentifierFields.reduce((object: any, field): A => object[field], actor);
+  protected getActorIdentifiers(actor: A): string[] | undefined {
+    const identifierValue = <string | string[] | undefined> this.actorIdentifierFields
+      .reduce((object: any, field): A => object[field], actor);
+    if (!identifierValue) {
+      return;
+    }
+    return Array.isArray(identifierValue) ? identifierValue : [ identifierValue ];
```

Comment:
Is it really required for this being possibly an array?

Because this logic will make it so that V8 can not make assumptions about the type of this value, which may slow down things.

Reply:
I think it does not matter here since this code is only run when an actor subscribes to the bus. It is thus only a startup issue?

Reply:
Ah wait, there's both `getActionIdentifier` and `getActionIdentifiers`.
The first one is used at runtime for finding actors.
The second one during registration.
If my understanding is correct, then this comment can be closed :-)

Reply:
Correct